src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java

```diff
@@ -790,20 +791,21 @@ private void storeItems() {
         final IMEInventory<IAEItemStack> ii = sg
                 .getInventory(Api.instance().storage().getStorageChannel(IItemStorageChannel.class));
 
-        for (IAEItemStack is : this.inventory.getItemList()) {
-            is = this.inventory.extractItems(is.copy(), Actionable.MODULATE, this.machineSrc);
-
-            if (is != null) {
-                this.postChange(is, this.machineSrc);
-                is = ii.injectItems(is, Actionable.MODULATE, this.machineSrc);
-            }
+        IItemList<IAEItemStack> itemList = this.inventory.getItemList();
+        for (IAEItemStack is : itemList) {
+            this.postChange(is, this.machineSrc);
+            IAEItemStack remainder = ii.injectItems(is.copy(), Actionable.MODULATE, this.machineSrc);
 
-            if (is != null) {
-                this.inventory.injectItems(is, Actionable.MODULATE, this.machineSrc);
+            // The network was unable to receive all of the items, i.e. no or not enough storage space left
+            if (remainder != null) {
+                is.setStackSize(remainder.getStackSize());
```

Comment:
I am not sure how this will behave in some extreme cases like size > `Integer.MAX_VALUE`. Crafting is certainly one of the few cases, which can run into these.

Reply:
This is all IAEItemStack, so it should be all longs?

Reply:
Yes, but we should clamp these to `Integer.MAX_VALUE` when they leave the network, like being stored in an external storage. And I do not think that is actually safeguarded at all.
This is usually fine as nothing else should move more than a few k at once. Crafting is really the only case where it can actually happen. But that might already apply for the current code.

Reply:
Yeah it was already stored in the CPU when it gets to this point, so I don't think this is any more "unsafe" than before, really.